Feature/dropdownMenu

Plofland opened the pull request:
## PR TITLE

**Description:** <!-- add description here -->
Sorry for the length of this PR, it got a bit out of hand.

The purpose of this PR is to put the discussion drawer header and discussion dropdown menu into their own components.

To do this I moved the flagging and moderator modal state to the redux store and all components using these slices of state have been updated.

The depreciated components (DCardDropdown & PopOverContent) were deleted along with their corresponding state in the redux store.

To avoid merge conflicts with Ismail's work on the comment system, I pulled down from the main and resolved the conflicts locally.
This may have still caused some problems because I'm not seeing added comments pop up in the discussionDrawer.
  <!-- What was changed? -->
  <!-- Why does PR exist? -->
  <!-- How do your changes do what is intended? -->
  <!-- Will a screenshot be helpful? -->

### Tests
Ran tests locally.
  <!-- Do we need tests? -->

### Reviewers

  <!-- 2 reviewers are required to review PR before mergeing to main -->

### Checklist

 * [x] My code follows the style guidelines of this project
 * [x] I have performed a self-review of my own code
 * [ ] My code has been reviewed by at least one peer
 * [x] I have commented my code, particularly in hard-to-understand areas
 * [ ] I have made corresponding changes to the documentation
 * [x] My changes generate no new warnings
 * [ ] I have added tests that prove my fix is effective or that my feature works
 * [x] New and existing unit tests pass locally with my changes
 * [ ] There are no merge conflicts

Plofland committed:
Copied over most DiscussionCard code to DiscussionDrawer

Plofland committed:
Adapted DiscussionCard logic for DiscussionDrawer

Plofland committed:
Added author, upvote, and dropdown controls to DiscussionDrawer

Plofland committed:
Moved pin/flag dropDown menu into its own component

Plofland committed:
Deleted DCardDropdown from assets and set up global state for dropDownMenu

Plofland committed:
Global state set for showModal and showFlagModel and commented out depreciated code

Plofland committed:
showModal and showFlagModal reducers and actions are functional

Plofland committed:
Bug fixed with DropdownMenu not working in DrawerHeader

Plofland committed:
Fully incorporated the DropdownMenu component into the DiscussionCard

Plofland committed:
Cleaned up useless code, comments and todo's

Plofland committed:
Changed all global showModal's to showModModal's for clarification

Plofland committed:
In RoomContent, changed showModal to showNewCardModal for clarity

Plofland committed:
Merged branch 'main' into feature/DropdownMenu branch and resolved conflicts with Ismail's work

Plofland committed:
Cleaned up code and left comments

ialkamal commented on package.json:
```diff
@@ -24,10 +24,11 @@
     "react-router-dom": "^5.2.0",
     "react-scripts": "^4.0.3",
     "redux": "^4.0.5",
+    "redux-logger": "^3.0.6",
```
Add redux logger as a dev dependency. 
`npm uninstall <package-name>`
and install as dev dependency
`npm install <package-name> --save-dev`

Plofland replied:
AH! Thank you. I moved it there because I was having issues with the redux logger but I never moved it back once it was fixed.

ialkamal commented on src/assets/DiscussionCard.js:
```diff
@@ -34,14 +35,14 @@ const IconText = ({ icon, text }) => (
+
+  // const flagsLength = props.discussion.flags
+  //   ? props.discussion.flags.length
+  //   : 0;
```
yes and the useEffect also as these are being handled by the Feed component

robo-chick reviewed (approved):
Nice work!

ialkamal reviewed (changes requested):
Since showing comments is critical, I would wait till you fix the issue.

Plofland committed:
Discussion Card old code

Removed old commented-out code that is no longer needed

Plofland committed:
Package.JSON logger

Moved redux-logger package from dependencies to devDependencies

Plofland committed:
Package.JSON

Rolled back the "react-redux": "^7.2.3" version to 7.2.0 to match the main branch
